cmd/private_network_create.go

```diff
@@ -48,37 +47,90 @@ func (c *privateNetworkCreateCmd) cmdPreRun(cmd *cobra.Command, args []string) e
 }
 
 func (c *privateNetworkCreateCmd) cmdRun(_ *cobra.Command, _ []string) error {
-	ctx := exoapi.WithEndpoint(gContext, exoapi.NewReqEndpoint(account.CurrentAccount.Environment, c.Zone))
+	ctx := gContext
+	client, err := switchClientZoneV3(ctx, globalstate.EgoscaleV3Client, c.Zone)
+	if err != nil {
+		return err
+	}
 
-	privateNetwork := &egoscale.PrivateNetwork{
-		Description: utils.NonEmptyStringPtr(c.Description),
-		EndIP: func() (v *net.IP) {
+	req := v3.CreatePrivateNetworkRequest{
+		Description: func() string {
+			if c.Description != "" {
+				return *utils.NonEmptyStringPtr(c.Description)
+			}
+			return ""
+		}(),
+		EndIP: func() net.IP {
 			if c.EndIP != "" {
-				ip := net.ParseIP(c.EndIP)
-				v = &ip
+				return net.ParseIP(c.EndIP)
 			}
-			return
+			return nil
 		}(),
-		Name: &c.Name,
-		Netmask: func() (v *net.IP) {
+		Name: c.Name,
+		Netmask: func() net.IP {
 			if c.Netmask != "" {
-				ip := net.ParseIP(c.Netmask)
-				v = &ip
+				return net.ParseIP(c.Netmask)
 			}
-			return
+			return nil
 		}(),
-		StartIP: func() (v *net.IP) {
+		StartIP: func() net.IP {
 			if c.StartIP != "" {
-				ip := net.ParseIP(c.StartIP)
-				v = &ip
+				return net.ParseIP(c.StartIP)
 			}
-			return
+			return nil
 		}(),
 	}
 
-	var err error
+	if len(c.Option) > 0 {
+		opts := &v3.PrivateNetworkOptions{}
+		optionsMap := make(map[string][]string)
+
+		// Process each option flag
+		for _, opt := range c.Option {
+			keyValue := strings.SplitN(opt, "=", 2)
+			if len(keyValue) != 2 {
+				continue
+			}
+			key := keyValue[0]
+			values := strings.Split(keyValue[1], " ")
+			optionsMap[key] = append(optionsMap[key], values...)
+		}
+
+		// Process collected values
+		for key, values := range optionsMap {
+			switch key {
+			case "dns-servers":
+				for _, v := range values {
+					if ip := net.ParseIP(v); ip != nil {
+						opts.DNSServers = append(opts.DNSServers, ip)
+					}
+				}
+			case "ntp-servers":
+				for _, v := range values {
+					if ip := net.ParseIP(v); ip != nil {
+						opts.NtpServers = append(opts.NtpServers, ip)
+					}
+				}
+			case "routers":
+				for _, v := range values {
+					if ip := net.ParseIP(v); ip != nil {
+						opts.Routers = append(opts.Routers, ip)
+					}
+				}
+			case "domain-search":
+				opts.DomainSearch = values
+			}
```

Comment:
Missing `default:` here for unrecognized option key.
Either print error or throw error (inline with my comment above).